lib/cog_api/client.ex

```diff
@@ -62,6 +63,13 @@ defmodule CogApi.Client do
   @callback rule_delete(String.t, %Endpoint{}) :: atom
   @callback rule_delete(String.t, %Endpoint{}) :: {atom, [String.t]}
 
+  @callback trigger_by_name(%Endpoint{}, String.t) :: {:ok, %Trigger{}} | {:error, term}
```

Comment:
Should this be `trigger_show_by_name` just to make it clear that this is the `show` option?